tests/test_unittest/test_gtc/common_utils.py

```diff
@@ -35,8 +35,7 @@ def visit_Node(self, node: concepts.Node, *, symbols: Set[str], refs: Set[str])
                     elif issubclass(type_, type_definitions.SymbolRef):
                         refs.add(getattr(node, name))
 
-            if not isinstance(node, traits.SymbolTableTrait):
```

Comment:
What's this about?

Reply:
The point of `undefined_symbol_list` is to create a list of results of calls (usually `Decl`s) for each ref name without a decl iterated over by `CollectSymbolsAndRefs`. The problem when making `HorizontalExecution` a `SymbolTableTrait` is that the collector stops at the `VerticalLoopSection`, so it never sees even a single ref to any API field, so each test ends up with errors about undefined references from the validator.

Reply:
Yes, that’s perfect. I left that check there to make sure that we update the handling of nested symbol tables correctly once we introduce some. But as `undefined_symbol_list` is only used at the stencil level, this new behavior should probably be always correct.